packages/mattermost-components/package.json

```diff
@@ -0,0 +1,58 @@
+{
+  "name": "@mattermost/components",
+  "module": "dist/index.esm.js",
+  "types": "dist/index.esm.d.ts",
+  "scripts": {
+    "build": "rollup -c",
+    "debug": "rollup -c"
+  },
+  "workspaces": [
```

Comment:
I think this can be removed. As long as we're installing all of these packages from the top level `npm install`, they can freely depend on each other without considering the other packages as workspaces. 

They'll need to also be added to dependencies (or dev/peer dependencies) though for that to work.

Reply:
I think it makes more sense to import them like this because then you are actually using the local code rather then the npm package code.

Reply:
I think this is another thing that npm sorts out for us if we don't run `npm install` from the subfolder. I'm pretty sure most of the examples I've seen just have `workspaces` defined in the root `package.json`, and `npm` sorts out linking all the inter-workspace dependencies.

I think that works because of how Node looks for node_modules in the current directory, then its parent, and so on, so since there's no `node_modules` in the `mattermost-components` folder because we didn't `npm install` here, then it'll walk up the tree until it finds the root `node_modules` which contains `@mattermost/types` as a symlink. reselect might be a problem since it's a hack, but I think that might be motivation to change how we modify reselect since [the patching method that mobile uses](https://github.com/mattermost/mattermost-mobile/tree/gekidou/patches) has felt better when I've used it before

Reply:
Yes I think you are right. Just tried and it seems to work fine.